// redux-form/es/createFieldProps.js
import _noop from 'lodash-es/noop';

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

import createOnBlur from './events/createOnBlur';
import createOnChange from './events/createOnChange';
import createOnDragStart from './events/createOnDragStart';
import createOnDrop from './events/createOnDrop';
import createOnFocus from './events/createOnFocus';


var processProps = function processProps(type, props, _value) {
  var value = props.value;

  if (type === 'checkbox') {
    return _extends({}, props, {
      checked: !!value
    });
  }
  if (type === 'radio') {
    return _extends({}, props, {
      checked: value === _value,
      value: _value
    });
  }
  if (type === 'select-multiple') {
    return _extends({}, props, {
      value: value || []
    });
  }
  if (type === 'file') {
    return _extends({}, props, {
      value: undefined
    });
  }
  return props;
};

var createFieldProps = function createFieldProps(getIn, name, _ref) {
  var asyncError = _ref.asyncError;
  var asyncValidating = _ref.asyncValidating;
  var blur = _ref.blur;
  var change = _ref.change;
  var _ref$defaultValue = _ref.defaultValue;
  var defaultValue = _ref$defaultValue === undefined ? '' : _ref$defaultValue;
  var dirty = _ref.dirty;
  var focus = _ref.focus;
  var format = _ref.format;
  var normalize = _ref.normalize;
  var parse = _ref.parse;
  var pristine = _ref.pristine;
  var props = _ref.props;
  var state = _ref.state;
  var submitError = _ref.submitError;
  var value = _ref.value;
  var _value = _ref._value;
  var syncError = _ref.syncError;

  var custom = _objectWithoutProperties(_ref, ['asyncError', 'asyncValidating', 'blur', 'change', 'defaultValue', 'dirty', 'focus', 'format', 'normalize', 'parse', 'pristine', 'props', 'state', 'submitError', 'value', '_value', 'syncError']);

  var asyncValidate = arguments.length <= 3 || arguments[3] === undefined ? _noop : arguments[3];

  var error = syncError || asyncError || submitError;
  var onChange = createOnChange(change, { normalize: normalize, parse: parse });
  var fieldValue = value == null ? defaultValue : value;

  return _extends({
    input: processProps(custom.type, {
      name: name,
      onBlur: createOnBlur(blur, { normalize: normalize, parse: parse, after: asyncValidate.bind(null, name) }),
      onChange: onChange,
      onDragStart: createOnDragStart(name, fieldValue),
      onDrop: createOnDrop(name, change),
      onFocus: createOnFocus(name, focus),
      value: format ? format(fieldValue) : fieldValue
    }, _value),
    meta: {
      active: state && !!getIn(state, 'active'),
      asyncValidating: asyncValidating,
      dirty: dirty,
      error: error,
      invalid: !!error,
      pristine: pristine,
      touched: !!(state && getIn(state, 'touched')),
      valid: !error,
      visited: state && !!getIn(state, 'visited')
    }
  }, props, custom);
};

export default createFieldProps;
